R/zchunk_L242.Bld_Inc_Elas_scenarios.R

```diff
@@ -73,6 +78,20 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios <- function(command, ...) {
           add_precursors("common/GCAM_region_names", "socioeconomics/A42.inc_elas",
                          "L102.pcgdp_thous90USD_Scen_R_Y") })
 
+    # L242.IncomeElasticity_bld_GCAM3: building sector income elasticity for GCAM 3.0 socioeconomics
+    # For the GCAM 3.0 scenario, calculate the per-capita GDP
+    L242.IncomeElasticity_bld_GCAM3 <- L102.gdp_mil90usd_GCAM3_R_Y %>%
+      left_join_error_no_match(L101.Pop_thous_GCAM3_R_Y, by = c("GCAM_region_ID", "year")) %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      transmute(region, year, pcgdp_90thousUSD = value.x / value.y) %>%
+      mutate(income.elasticity = approx(x = A42.inc_elas$pcgdp_90thousUSD, y = A42.inc_elas$inc_elas,
```

Comment:
Ugh, does seem messy. OK maybe just add a comment explaining better why we're doing it this way. Thanks.